### ISSUE TITLE
Vanish not dropping aggro from mob. 

### ISSUE THREAD

**Issue by temperrr** [azerothcore/azerothcore-wotlk#7125]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [x] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Using vanish did not break mob aggro. As seen on the screens below, after using vanish they kept me in combat and just stood in place. I tried it on these Shadow mages and it bugged twice after trying it 4 times. I had this happen once before this in a low level dungeon, so I didnt think much of it, maybe its related to npc with pets.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

From wowwiki : "All aggroed mobs on the rogue will exit combat and return to their original locations or turn to other players if the rogue is in a party."

##### SOURCE:

 https://wowwiki-archive.fandom.com/wiki/Vanish
##### STEPS TO REPRODUCE THE PROBLEM:

1.  Run to the mob and get aggro
2. Start running away and use vanish
3. 
4.

##### EXTRA NOTES:

This is the NPC - https://wotlkdb.com/?npc=5617
Screens:
[https://imgur.com/a/AfaExI3](url)

Originally reported : https://github.com/chromiecraft/chromiecraft/issues/1181
Triager note: bug is pretty rare. Managed to get it to work after 20 or so tries (used vanish rank 1 and .cheat cooldown)

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/7c21d567d59153b37978756a5bf02daf6998e665

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by UltraNix** [comment]
Need more info about how to reproduce it. I cannot do it on my local.

**Comment by FrancescoBorzi** [comment]
> Need more info about how to reproduce it. I cannot do it on my local.

@temperrr can you please help?

**Comment by Azcobu** [comment]
Checking this one now, although still struggling to find a coherent way of reproducing it.

Ultra, is there some sort of mechanic built in to Vanish that forces NPCs to move directly away from you for a second or two? Because what I'm seeing from testing this is that usually when Vanish doesn't seem to work, it's actually working properly and the NPC just isn't moving far enough away, so they immediately re-detect you. But I'm curious about what makes the NPC move away from you and if there's some sort of minimum move distance you could set? Or doesn't it work like that at all?

**Comment by temperrr** [comment]
Vanish should give you a 10 second improved stealth, I wonder if that does get applied correctly. (not sure how to test this). I wonder if what Azcobu found with it just having to do with being to close to their reset point...

**Comment by Azcobu** [comment]
Yeah, I think it is caused by being too close to the reset point, so was wondering how the mechanic that makes NPCs move away from you when you hit Vanish worked, and if it had a minimum distance or something. If you could force the NPC to always move at least X yards away from you when you Vanish, instead of it just returning to a respawn spot that you might be standing at as well, the problems shown above wouldn't happen.

But I'm still expecting Ultra to tell us it doesn't work like that anyway, so it's probably a moot point.

**Comment by UltraNix** [comment]
When you vanish, then mob just enters evade mode. And when it moves to its home it can detect you again.

**Comment by Azcobu** [comment]
Is there a way to force a minimum evade movement distance? Say if you're standing on its home position when you Vanish, can you force it to move 5-10 yards away anyway?

**Comment by UltraNix** [comment]
No - evade is moving to its home position.

**Comment by Azcobu** [comment]
Could you define a new home position if it was too close to the player and make the NPC move there instead?

**Comment by UltraNix** [comment]
It would looks weird and it's not blizzlike for sure :P

**Comment by Shineslippers** [comment]
May i say about something similar. Maybe this is the case. Example:

https://user-images.githubusercontent.com/85256266/130300099-ea6930bb-0623-4a87-b0ec-24379dc07e85.mp4

Recap. 1st NPC technicaly grabbed 2nd NPC before my vanish. But there is existing delay between moment when 2nd NPC was grabbed and moment when 2nd NPC started move. If someone pressing vanish between this moments we're getting situation when you can say "vahish doesn't work".

**Comment by UltraNix** [comment]
Yep - there is a call assistance mechanic here. There is no check if that grabbed mob can attack you after that delay. But this is only one part of the problems.

**Comment by UltraNix** [comment]
#4764 is about the same.
